Cargo.toml

```diff
@@ -2,3 +2,6 @@
 members = [
     "chacha20poly1305",
 ]
+
+[patch.crates-io]
+aead = { git = "https://github.com/RustCrypto/traits" }
```

Comment:
@newpavlov I just removed `[WIP]` as this PR is functionally complete. Not sure how you feel about merging this with ^^^, but otherwise I'd call this PR ready for review

Reply:
I think it would be better to depend on git for `aead` directly in the `chacha20poly1305` crate. But overall I don't have any objections against temporarily using a git dependency. :)

Reply:
Cargo doesn't allow you to patch dependencies at the individual crate level within a workspace. I imagine that precludes a workspace-wide dependency resolution.

Reply:
I meant not patching, but using `aead = { git = "https://github.com/RustCrypto/traits" }` instead of `aead = "0.1"` in the `[dependencies]` section.